core/rawdb/accessors_chain.go

```diff
@@ -573,14 +573,12 @@ func ReadCXReceiptsProofUnspent(db DatabaseReader, shardID uint32, number uint64
 	return common.BytesToHash(data), nil
 }
 
-// TryWriteCXReceiptsProofUnspent check whether a CXReceiptsProof is unspent, write to database if unspent
-func TryWriteCXReceiptsProofUnspent(dbr DatabaseReader, dbw DatabaseWriter, shardID uint32, number uint64, blockHash common.Hash) error {
-	hash, _ := ReadCXReceiptsProofUnspent(dbr, shardID, number)
-	// only write to database for the first time given a specified block number
-	if hash == EmptyHash {
-		return dbw.Put(cxReceiptUnspentKey(shardID, number), blockHash.Bytes())
-	}
-	return ctxerror.New("[TryWriteCXReceiptsProofUnspent] blockHash already exist", "had", hash, "tryPut", blockHash)
+// WriteCXReceiptsProofUnspent check whether a CXReceiptsProof is unspent, write to database if unspent
```

Comment:
The code doesn’t check it’s unspent or not.

Reply:
Yes, this is only write to database, the low level function. The checking is done in blockchain.go

Reply:
comments modified